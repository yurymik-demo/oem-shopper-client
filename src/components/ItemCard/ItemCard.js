import React from "react";

const ItemCard = props => {
  const { title, description, price } = props.item;

  return (
    <div className="row item-card">
      <div className="col s12 m12">
        <div className="card blue-grey darken-1">
          <div className="card-content white-text">
            <span className="card-title">{title}</span>
            <p>{description}</p>
          </div>
          <div className="card-action">
            <a href="#">Ratings</a>
            <div className="right white-text">{price}</div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ItemCard;
