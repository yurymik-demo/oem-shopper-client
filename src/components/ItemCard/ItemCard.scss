.item-card {
  .card {
    .card-action {
      min-height: 50px;

      // Need to override materialize css, and trying not to have to use !important
      a:not(.btn):not(.btn-large):not(.btn-small):not(.btn-large):not(.btn-floating).link.view-details-action {
        margin-right: 0;
      }
    }
  }
}
