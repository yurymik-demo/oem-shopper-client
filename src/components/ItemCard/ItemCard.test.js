import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import ItemCard from "./ItemCard";

it("renders without crashing", () => {
  const mockItem = {
    title: "title",
    description: "item descriotion",
    price: "$123.00"
  };
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <ItemCard item={mockItem} />
    </BrowserRouter>,
    div
  );
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});
