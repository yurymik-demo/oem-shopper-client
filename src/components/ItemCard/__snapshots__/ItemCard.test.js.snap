// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="row item-card"
  >
    <div
      class="col s12 m12"
    >
      <div
        class="card white "
      >
        <div
          class="card-content "
        >
          <span
            class="card-title"
          >
            title
          </span>
          <div
            class="right"
          >
            $
            123.00
          </div>
          <p>
            item descriotion
          </p>
        </div>
        <div
          class="card-action"
        >
          <div
            class="item-rating-container left"
          >
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star
              </i>
            </a>
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star
              </i>
            </a>
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star
              </i>
            </a>
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star_border
              </i>
            </a>
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star_border
              </i>
            </a>
          </div>
          <a
            class="right"
            href="/items/undefined"
          >
            View Details
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
